File: lyedu-api/src/main/java/com/lyedu/entity/Knowledge.java
package com.lyedu.entity;

import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.List;

/**
 * 知识库实体
 *
 * @author LyEdu Team
 */
@Data
@EqualsAndHashCode(callSuper = true)
public class Knowledge extends BaseEntity {

    /**
     * 标题/名称
     */
    private String title;

    /**
     * 分类（如：制度文档、技术文档）
     */
    private String category;

    /**
     * 文件名
     */
    private String fileName;

    /**
     * 文件地址
     */
    private String fileUrl;

    /**
     * 文件大小（字节）
     */
    private Long fileSize;

    /**
     * 文件类型/扩展名
     */
    private String fileType;

    /**
     * 排序
     */
    private Integer sort;

    /**
     * 可见性：1-公开，0-私有（按部门）
     */
    private Integer visibility;

    /**
     * 关联部门ID列表（私有时；不持久化到 ly_knowledge，来自 ly_knowledge_department）
     */
    private List<Long> departmentIds;
}

File: lyedu-admin/src/api/knowledge.ts
import request from '@/utils/request'

export interface Knowledge {
  id: number
  title: string
  category?: string
  fileName?: string
  fileUrl: string
  fileSize?: number
  fileType?: string
  sort: number
  /** 可见性：1-公开，0-私有 */
  visibility: number
  departmentIds?: number[]
  createTime?: string
  updateTime?: string
}

export interface PageResult<T> {
  records: T[]
  total: number
  current: number
  size: number
  pages: number
}

export const getKnowledgePage = (params: {
  page: number
  size: number
  keyword?: string
  category?: string
}) => {
  return request.get<PageResult<Knowledge>>('/knowledge/page', { params })
}

export const getKnowledgeByIdAdmin = (id: number) => {
  return request.get<Knowledge>(`/knowledge/admin/${id}`)
}

export const createKnowledge = (data: {
  title: string
  category?: string
  fileName?: string
  fileUrl: string
  fileSize?: number
  fileType?: string
  sort?: number
  visibility?: number
  departmentIds?: number[]
}) => {
  return request.post<number>('/knowledge', data)
}

export const updateKnowledge = (id: number, data: {
  title: string
  category?: string
  fileName?: string
  fileUrl: string
  fileSize?: number
  fileType?: string
  sort?: number
  visibility?: number
  departmentIds?: number[]
}) => {
  return request.put(`/knowledge/${id}`, data)
}

export const deleteKnowledge = (id: number) => {
  return request.delete(`/knowledge/${id}`)
}
